enhancements/20230316-baremetal-container-mcm-support.md

```diff
@@ -0,0 +1,151 @@
+   - the local `/v1/harvester` API endpoint is only available starting from the v1.2 after enabling the multi-cluster-management feature in the Rancher server config.
+
+
+## Design
+
+### Single Harvester UI
+
+The Single UI will remain mostly unchanged, and users can still access the embedded Rancher explorer UI when the 'rancher-manager-support' setting is disabled.
+
+### Multi-cluster UI Design Change
+
+1. Add the `Harvester Dashboard` menu at the top of the `Home` menu to allow users to visit the local Harvester dashboard page easily(local Harvester mode only).
+![Harvester-Dashboard-memu.png](./20230316-baremetal-container-mcm-support/harvester-dashboard.png)
+2. Virtualization Management pages
+    - Change the `version` column value on the `Virtualization Management` page to the Harvester version.(require backend support)
+    ![harvester-version.png](./20230316-baremetal-container-mcm-support/harvester-version.png)
+    - Add a `Set as login page` option to the header of the Harvester dashboard page.
+5. Show the Harvester clusters in the `Cluster Management` & `Continuous Delivery` & `Explore cluster` list. In the `EXPLORE CLUSTER` list, if it is a Harvester cluster, we need to replace it with the Harvester icon.
+6. Global setting(local Harvester mode only):
+   - Sorting out global setting and harvester settings, remove/hide duplicate settings(require backend research)
+   - Lock the `harvester` and `multi-cluster-management` in the `Feature Flags` list like rke2 (can not be disabled)
+7. Home page
+   - Allow users to jump from the home page to the Harvester dashboard by adding a Harvester icon to the right of the name. Clicking on the icon will take the user to the Harvester Dashboard page.
+   - Show all clusters (including Harvester) on the home page.
+   - The current local cluster's provider is shown as Custom, but it should be shown as Harvester, including the cluster management table page. See the screenshot below:
+   ![home-icon.png](./20230316-baremetal-container-mcm-support/home-icon.png)
+8. Cluster Management page:
+   - Add the `Explore` button to the end of the local cluster.
+9. Explorer UI of local Harvester cluster
+   - Disable/hide the Rancher upgrade button on the `Install Apps` page.
+   ![rancher-upgrade-button.png](./20230316-baremetal-container-mcm-support/rancher-upgrade-button.png)
+   - Disable edit/upgrade/delete Harvester managed system service (e.g., monitoring, logging, Harvester...)
+    ![harvester-service.png](./20230316-baremetal-container-mcm-support/harvester-service.png)
+   - In the `Cluster Tools` page, we should disable operations that could affect Harvester, such as edit/delete monitor, longhorn, and logging.
+    ![cluster-tools.png](./20230316-baremetal-container-mcm-support/cluster-tools.png)
+   - If the `local` cluster is a harvester cluster, we need to hide resources prefixed with `harvesterhci.io.*`, and `kubevirt.io.*`
```

Comment:
Why do we need to hide those resources?

Reply:
Currently, there are two main reasons:
1. we are already showing those essential resources on the virtualization page. Also, the container view of those resources is not well designed...
2. operations with system-level CRDs, e.g., kubevirt CR could potentially break the system behavior and block it from successfully upgrade when there is a conflict.

Reply:
How does it handle Longhorn or other Rancher-related CRDs? We can follow suit.

Reply:
We have proposed to disable the operations(edit/upgrade/delete) against Harvester managed system service (e.g., monitoring, logging, Harvester, Longhorn...) when it is the local cluster's provider is Harvester.

Reply:
I meant if a user installs Longhorn on Rancher (outside of Harvester), can they see those CRs?

Reply:
This will only impact the `local` cluster with its provider being `Harvester`.